core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/WorkspaceMigrationMonitor.scala

```diff
@@ -298,22 +174,44 @@ object V1WorkspaceMigrationMonitor
     IO.raiseError(new IllegalStateException(s"""Failed to retrieve workspace bucket "${name}"."""))
 
 
-  final def failNoGoogleProject[A](attempt: V1WorkspaceMigrationAttempt): IO[A] =
+  final def failNoGoogleProject[A](attempt: WorkspaceMigration): IO[A] =
     IO.raiseError(new IllegalStateException(s"""Google Project was not created for migration "${attempt.id}.""""))
 
 
-  final def failNoTmpBucket[A](attempt: V1WorkspaceMigrationAttempt): IO[A] =
+  final def failNoTmpBucket[A](attempt: WorkspaceMigration): IO[A] =
     IO.raiseError(new IllegalStateException(s"""Temporary storage bucket was not created for migration "${attempt.id}.""""))
 
+
+  final def startBucketStorageTransferJob(migration: WorkspaceMigration,
+                                          originBucket: GcsBucketName,
+                                          destBucket: GcsBucketName,
+                                          projectToBill: GoogleProject,
+                                          sts: GoogleStorageTransferService[IO]
+                                          ): IO[(TransferJob, WriteAction[Unit])] =
```

Comment:
I'm starting to think that this return type is not what we want. Returning the `DBIOAction` separately means we aren't guaranteed to write the result of the STS operation (or any other GCP operations) to the DB immediately after they complete -- it leaves it up to whoever is calling this fn to be diligent about running the `DBIOAction` right after and leaves open the possibility that they won't run it at all which we definitely don't want. The alternative is to wrap all calls to the `MigrationMonitor` in `inTransaction` blocks, but that's not ideal either since a lot of those calls involve issuing requests to GCP which would tie up DB connections. 

My proposed solution would be to run the `DBIOAction` in this function and have the return type just be `IO[TransferJob]`.